src/network/postgres_protocol_handler.cpp

```diff
@@ -1056,6 +1059,87 @@ bool PostgresProtocolHandler::ReadPacket(Buffer &rbuf, InputPacket &rpkt) {
   return true;
 }
 
+/*
+ * process_startup_packet - Processes the startup packet
+ *  (after the size field of the header).
+ */
+bool PostgresProtocolHandler::ProcessInitialPacket(InputPacket *pkt, Client client, bool ssl_able, bool& ssl_handshake, bool& finish_startup_packet) {
+  std::string token, value;
+  std::unique_ptr<OutputPacket> response(new OutputPacket());
+
+  int32_t proto_version = PacketGetInt(pkt, sizeof(int32_t));
+  LOG_INFO("protocol version: %d", proto_version);
+
+  // TODO: consider more about return value
+  if (proto_version == SSL_MESSAGE_VERNO) {
+    LOG_TRACE("process SSL MESSAGE");
+    return ProcessSSLRequestPacket(pkt, ssl_able, ssl_handshake);
+  }
+  else {
+    LOG_TRACE("process startup packet");
+    return ProcessStartupPacket(pkt, proto_version, client, finish_startup_packet);
+  }
+}
+
+bool PostgresProtocolHandler::ProcessSSLRequestPacket(InputPacket *pkt, bool ssl_able, bool& ssl_handshake) {
+  UNUSED(pkt);
```

Comment:
@tli2 It is unnecessary since the SSL startup packet only contains the version number for SSL, which is already checked outside the call. I will just remove the parameter from the function prototype.